backend/corpora/api_server/logger.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+from logging import LogRecord
+from logging.config import dictConfig
+from typing import List
+
+from pythonjsonlogger import jsonlogger
+from pythonjsonlogger.jsonlogger import merge_record_extra
+
+
+# The fields to log using the json logger.
+LOGGED_FIELDS = ["levelname", "asctime", "name", "message"]
+LOG_FORMAT = " ".join([f"%({field})" for field in LOGGED_FIELDS])
+
+
+class JsonFormatter(jsonlogger.JsonFormatter):
```

Comment:
Instead of unioning `LOGGED_FIELDS` and additional fields, it might be cleaner to _nest_ the additional fields as the `message` and add a log msg `type` property. This would allows for different "flavors" of log messages to be recorded, beyond just API logging.

```
{
  "level": "<LEVEL>",
 "asctime": "<TIME>",
 "type": "REQUEST" | "METRIC" | "QUERY" | etc
 "msg": { <JSON> }
}
```
When parsing log programmatically, one could filter on `type`, and then know how to reliably parse the `msg` JSON content.

Reply:
I moved the log specific details to `details`